akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -699,6 +703,16 @@ private[akka] class ShardRegion(
     retryCount = 0
   }
 
+  def deliverStartEntity(msg: StartEntity, snd: ActorRef): Unit = {
+    try {
+      deliverMessage(msg, snd)
+    } catch {
+      case ex: MatchError ⇒
+        log.error(ex, "When using remember-entities the shard id extractor must handle ShardRegion.StartEntity(id).")
+      // TODO fail sharding entirely?
```

Comment:
failing by throwing when registering the entity type would be nice, but that would mean that we would have to call their extractor function with a dummy id and that might not be safe. After that I don't see how we can "fail" sharding apart from logging. shutting down actor system would be too much